tests/backup/backup_psa_test.go

```diff
@@ -943,3 +944,460 @@ var _ = Describe("{RestoreFromHigherPrivilegedNamespaceToLower}", Label(TestCase
+				mutex.Lock()
+				scheduleNames = append(scheduleNames, schBackupName)
+				mutex.Unlock()
+				labelSelectors := make(map[string]string)
+				baselineScheduleBackup, err = CreateScheduleBackupWithValidation(ctx, schBackupName, SourceClusterName, backupLocationName, backupLocationUID, baselineScheduledAppContexts, labelSelectors, BackupOrgID, preRuleName, preRuleUid, postRuleName, postRuleUid, periodicPolicyName, schPolicyUid)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of schedule backup %v with baseline namespaces using exec rules", baselineScheduleBackup))
+				mutex.Lock()
+				schBackupList = append(schBackupList, baselineScheduleBackup)
+				mutex.Unlock()
+				log.InfoD("Schedule backup list after schedule backup with baseline namespaces - %v", schBackupList)
+			})
+		}()
+		wg.Wait()
+
+		Step("Restore all namespace backup to destination cluster with namespace & project mappings", func() {
+			log.InfoD("Restore all namespace backup to destination cluster with namespace & project mappings")
+			log.Infof("All namespace backup which is going to restore is %v", allNSScheduleBackup)
+			restoreName := fmt.Sprintf("restore-all-%v", RandomString(5))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestoreOnRancherWithoutCheck(restoreName, allNSScheduleBackup, destNamespaceProjectMapping, DestinationClusterName, BackupOrgID, ctx, nil, restoreProjectUIDMapping, restoreProjectNameMapping, 2)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of restore with replace option %s from backup %s", restoreName, allNSScheduleBackup))
+		})
+
+		Step("Custom restore of restricted namespaces to destination cluster", func() {
+			log.InfoD("Custom restore of restricted namespaces to destination cluster")
+			restoreName := fmt.Sprintf("restore-restricted-%v", RandomString(5))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestoreWithProjectMapping(restoreName, restrictedScheduleBackup, destNamespaceProjectMapping, DestinationClusterName, BackupOrgID, ctx, nil, restoreProjectUIDMapping, restoreProjectNameMapping)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Custom restore %s from restricted namespace backup %s", restoreName, restrictedScheduleBackup))
+		})
+
+		Step("Default restore of baseline namespaces to destination cluster", func() {
+			log.InfoD("Default restore of baseline namespaces to destination cluster")
+			restoreName := fmt.Sprintf("restore-baseline-%v", RandomString(5))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestoreWithValidation(ctx, restoreName, baselineScheduleBackup, make(map[string]string), make(map[string]string), DestinationClusterName, BackupOrgID, baselineScheduledAppContexts)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Default restore %s from baseline namespace backup %s", restoreName, baselineScheduleBackup))
+		})
+
+	})
+
+	JustAfterEach(func() {
+		log.Infof("Cleaning up the resources after the test")
+		defer func() {
```

Comment:
The defer should come in JustBeforeEach itself. refer : https://github.com/portworx/torpedo/blob/e54bd7a57d95cddd1cd9998801f7ce887fb0ca3f/tests/backup/backup_partial_test.go#L55C3-L55C19

cc: @mkoppal-px